plugins/helo.checks.js

```diff
@@ -41,6 +40,17 @@ exports.register = function () {
     this.register_hook('helo', 'emit_log');
     this.register_hook('ehlo', 'emit_log');
 
+    // IP literal that doesn't match remote IP
+    this.register_hook('helo', 'literal_mismatch');
+    this.register_hook('ehlo', 'literal_mismatch');
+
+    if (this.cfg.check.literal_mismatch === undefined) {
```

Comment:
this seems like a good use of the nullish coalescing operator (??).
```suggestion
    this.cfg.check.literal_mismatch === this.cfg.check.literal_mismatch ?? false;
```

Reply:
I've found more issues but I don't know how to cancel this pull request.
I create a new PR once I get these issues resolved.

Reply:
You can just push additional commit to your branch lnedry:patch-4 and the PR will be updated automatically.

Reply:
The first issue that I fixed was the default value for this.cfg.check.literal_mismatch. It was set to true when it should have had a value of 2.

Reply:
I found more issues. None of the hooks were getting registered.  There was a for...in loop which loops over the checks array. To get the correct values from the array, it should be a for...of loop. 

Fixing this issue gets the hooks registered, but now each hook get called twice. So now the first time a hook gets called, I check for a previously added result.